src/main/java/sdu/sem2/se17/persistence/data/CreditHandlerImpl.java

```diff
@@ -4,43 +4,163 @@
 import sdu.sem2.se17.domain.persistenceinterface.CreditHandler;
 import sdu.sem2.se17.persistence.db.DataSource;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.Optional;
 
 public class CreditHandlerImpl implements CreditHandler {
     private final DataSource dataSource;
+    private ParticipantHandlerImpl participantHandler;
```

Comment:
You should target the interface and not the implementation, all other places where this is an issue should also be fixed.
```suggestion
    private ParticipantHandler participantHandler;
```